app/Http/Controllers/AdminAuthController.php

```diff
@@ -5,23 +5,31 @@
 use App\Models\Admin;
 use Illuminate\Http\Request;
 use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Validator;
 
 class AdminAuthController extends Controller
 {
     public function register(Request $request)
     {
-        $fields = $request->validate([
-            'first_name' => ['required', 'string', 'max:40', 'regex:/(^([a-zA-Z]+)(\d+)?$)/u'],
-            'last_name' => ['required', 'string', 'max:40', 'regex:/(^([a-zA-Z]+)(\d+)?$)/u'],
+        $validator = Validator::make($request->all(), [
+
+            'first_name' => ['required', 'string', 'max:20', 'regex:/(^([a-zA-Z]+)?$)/u'],
+            'last_name' => ['required', 'string', 'max:20', 'regex:/(^([a-zA-Z]+)?$)/u'],
             'email' => ['required', 'string', 'unique:users,email', 'email', 'max:40'],
             'password' => ['required', 'string', 'confirmed', 'max:40'],
         ]);
```

Comment:
Didn't we agree on using `$this->request` ?

Reply:
yes, but I had to do it like that to send the error messages.

Reply:
You are using 
```php
'errors' => $validator->messages(),
```
Which basically what `$this->validate` do